added is_closed def to mesh_slice_plane

nizartaha opened the pull request:
<!-- Thank you for your pull request!  -->
<!-- Please start by describing your change in a few sentences. -->
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

nizartaha committed:
added is_closed property

checks if the mesh is closed. if its closed, it adds faces from boundary vertices. if its open it does not add these faces to the mesh.

nizartaha committed:
different method name

nizartaha committed:
update changelog

nizartaha committed:
remove white space

tomvanmele commented:
@nizartaha do i understand correctly that the goal is to return closed meshes if the original mesh was closed?

nizartaha commented:
@brgcode correct :)

and not to close it, if the original mesh was open. here is a quick test i did before i added `is_mesh_closed` check:
If the line `mesh.add_face(mesh.vertices_on_boundary())` was commented out in both `positive `and `negative` def, it keeps it open, which is needed here.

however, if this line was not commented out in one or both of them, it will just close it with strange faces from the boundary. which is not needed here.

I hope it is clear.


![image](https://user-images.githubusercontent.com/38039580/134002729-27070262-832d-4335-b05e-c4733659fc14.png)

nizartaha committed:
fix intersection ValueError

checks if intersection_segment_plane returns the mesh vertex (u or v of the edge ). 
if True appends the mesh vertex itself to intersections instead of calculating t. otherwise when mesh.split_edge is executed, it raises ValueError: t should be greater than 0.0. or ValueError: t should be smaller than 1.0.

tomvanmele committed:
Merge branch 'main' into main
